src/components/Facepile/Facepile.tsx

```diff
@@ -1,62 +1,144 @@
         </div>
       </div>
     );
   }
 
+  private _getPersonasToDisplay(personas: IFacepilePersona[]): IFacepilePersona[] {
+    let numPersonasToShow: number = this._calulateNumPersonasToShow();
+    let numPersonasNotPictured: number = this.props.personas.length - numPersonasToShow;
+    let hasPersonasNotPictured: boolean = numPersonasNotPictured > 0;
+    let personasToShow: IFacepilePersona[] = this.props.personas.slice(0, numPersonasToShow);
+
+    if (this.props.overflowPersonaProps) {
+      this._addNumberNotPictured(numPersonasToShow, hasPersonasNotPictured, numPersonasNotPictured, personasToShow);
+    }
+
+    return personasToShow;
+  }
+
+  private _addNumberNotPictured(numPersonasToShow: number, hasPersonasNotPictured: boolean, numPersonasNotPictured: number, personasToShow: IFacepilePersona[]): void {
+    let overflowPersonaProps: IFacepilePersona = this.props.overflowPersonaProps ? this.props.overflowPersonaProps : {};
+    if (hasPersonasNotPictured) {
+      personasToShow.push({
+        imageInitials: overflowPersonaProps.imageInitials || '+' + numPersonasNotPictured,
+        imageUrl: overflowPersonaProps.imageUrl,
+        initialsColor: overflowPersonaProps.initialsColor || PersonaInitialsColor.black,
+        personaName: overflowPersonaProps.personaName || this.props.personas.slice(numPersonasToShow).map((persona: IFacepilePersona, index: number) => {
+          return persona.personaName;
+        }).join(', '),
+        onClick: overflowPersonaProps.onClick,
+        onMouseMove: overflowPersonaProps.onMouseMove,
+        onMouseOut: overflowPersonaProps.onMouseOut
+      });
+    }
+  }
+
+  private _calulateNumPersonasToShow(): number {
+    let maxShownPersonas: number = this.props.maxDisplayablePersonas;
+
+    // Remove one for the add person button
+    if (this.props.showAddButton) {
+      --maxShownPersonas;
+    }
+
+    // Remove one if max exceeded for the +1 icon
+    if (this.props.overflowPersonaProps && this.props.personas.length > maxShownPersonas) {
+      --maxShownPersonas;
+    }
+
+    return this.props.personas.length < maxShownPersonas ? this.props.personas.length : maxShownPersonas;
+  }
+
   private _getPersonaControl(persona: IFacepilePersona): JSX.Element {
+    let personalDetailsHidden: boolean = this.props.personas.length > 1;
     return  <Persona
               imageInitials={ persona.imageInitials }
               imageUrl={ persona.imageUrl }
               initialsColor={ persona.initialsColor }
               primaryText={ persona.personaName }
               size={ PersonaSize.extraSmall }
-              hidePersonaDetails={ true } />;
+              hidePersonaDetails={ personalDetailsHidden } />;
   }
 
   private _getElementWithOnClickEvent(personaControl: JSX.Element, persona: IFacepilePersona, index: number): JSX.Element {
     return  <button
+              { ...getNativeProps(persona, buttonProperties) }
```

Comment:
Moved this up top as it is overriding the properties like onClick when it's placed after it.